src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionMetadataProvider.cs

```diff
@@ -50,6 +50,14 @@ public static class ExpressionMetadataProvider
                     // Property/field access is always legal
                     var memberExpression = (MemberExpression)expression.Body;
                     propertyName = memberExpression.Member is PropertyInfo ? memberExpression.Member.Name : null;
+                    if (string.Equals(propertyName, "Model", StringComparison.Ordinal) &&
```

Comment:
Yeah, we discussed this and their aren't great solutions. If we assume that you meant `m => m` when you meant `m => m.Model` the worst case is that we lose metadata on the `m.Model` property.

It's unfortunate we can't be completely precise, but this seems ok.